lib/src/components/NxTransferListHalf/__tests__/NxTransferListHalf.test.tsx

```diff
@@ -4,495 +4,537 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import { faArrowDown, faArrowUp, faEdit, faPlusCircle, faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+// import { faArrowDown, faArrowUp, faEdit, faPlusCircle, faTimesCircle } from '@fortawesome/free-solid-svg-icons';
 import { includes } from 'ramda';
+import { faEdit } from '@fortawesome/free-solid-svg-icons';
 import React from 'react';
+import { rtlRenderElement, rtlRender } from '../../../__testutils__/rtlUtils';
+import { within, render, act, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
-import { getShallowComponent, getMountedComponent } from '../../../__testutils__/enzymeUtils';
-import NxTooltip from '../../NxTooltip/NxTooltip';
-import NxFieldset from '../../NxFieldset/NxFieldset';
-import NxFilterInput from '../../NxFilterInput/NxFilterInput';
 import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
-
-import NxTransferListHalf from '../NxTransferListHalf';
-import { Props } from '../types';
+import NxTransferListHalf, { Props } from '../NxTransferListHalf';
 
 describe('NxTransferListHalf', function() {
-  const minimalProps = {
+  const minimalProps: Props = {
         label: 'Foo',
         filterValue: '',
         onFilterChange: () => {},
         items: [],
         footerContent: <div/>
       },
-      getShallow = getShallowComponent<Props<number>>(NxTransferListHalf, minimalProps),
-      getMounted = getMountedComponent<Props<number>>(NxTransferListHalf, minimalProps);
+      quickRender = rtlRender(NxTransferListHalf, minimalProps),
+      renderEl = rtlRenderElement(NxTransferListHalf, minimalProps);
+
+  beforeEach(function() {
+    // JSDOM is missing this function. https://github.com/jsdom/jsdom/issues/3002
```

Comment:
Found the same solution online as `NxBreadCrumb` after getting `range.getBoundingClientRect is not a function`.

Reply:
> Found the same solution online as NxBreadCrumb after getting range.getBoundingClientRect is not a function.

It's seeming like that's going to be necessary everywhere that `NxOverflowTooltip` is used.  Rather than having to mock that everywhere, that mock should be moved into `setupTests.ts`.  That file is configured using jest's [`setupFilesAfterEnv`](https://jestjs.io/docs/configuration#setupfilesafterenv-array)

Reply:
> It's seeming like that's going to be necessary everywhere that `NxOverflowTooltip` is used. Rather than having to mock that everywhere, that mock should be moved into `setupTests.ts`. That file is configured using jest's [`setupFilesAfterEnv`](https://jestjs.io/docs/configuration#setupfilesafterenv-array)

Moved into `setupTests.ts` here: [26ae3c2](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/26ae3c2df710029d6a89f10da24d594df39338e9). I kept getting an eslint error for `jest.fn()` saying `jest is not defined`. Andrew had a good suggestion of adding `setupTests.js` to the `.eslintrc` file in order to deal with this. If all looks good, then I'll also remove the same `beforeEach` in NxBreadcrumb.

Reply:
> Moved into setupTests.ts here: [26ae3c2](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/26ae3c2df710029d6a89f10da24d594df39338e9). I kept getting an eslint error for jest.fn() saying jest is not defined. Andrew had a good suggestion of adding setupTests.js to the .eslintrc file in order to deal with this. If all looks good, then I'll also remove the same beforeEach in NxBreadcrumb.

Great.  I also made similar changes in RSC-1336.  You'll want to merge that branch into this one so you can more cleanly deal with the timing problems.

Reply:
> > Moved into setupTests.ts here: [26ae3c2](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/26ae3c2df710029d6a89f10da24d594df39338e9). I kept getting an eslint error for jest.fn() saying jest is not defined. Andrew had a good suggestion of adding setupTests.js to the .eslintrc file in order to deal with this. If all looks good, then I'll also remove the same beforeEach in NxBreadcrumb.
> 
> Great. I also made similar changes in RSC-1336. You'll want to merge that branch into this one so you can more cleanly deal with the timing problems.

I merged your changes here: [5eed414](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/5eed41445b5e905b6c731b3ef177428f780c46e1)